contracts/src/cryptography/message_hash_utils.rs

```diff
@@ -0,0 +1,93 @@
+//! Signature message hash utilities for producing digests to be consumed by
+//! `ECDSA` recovery or signing.
+//!
+//! The library provides methods for generating a hash of a message that
+//! conforms to the [ERC-191] and [EIP 712] specifications.
+//!
+//! [ERC-191]: https://eips.ethereum.org/EIPS/eip-191
+//! [EIP-712]: https://eips.ethereum.org/EIPS/eip-712
+
+use alloc::string::String;
+
+use alloy_primitives::{bytes, keccak256, B256};
+use alloy_sol_types::{sol, SolType};
+
+type EthMessageCoder = sol! {
+    tuple(bytes, bytes32)
+};
+
+type TypeHashCoder = sol! {
+    tuple(string, bytes32, bytes32)
+};
```

Comment:
I think it's better to co-locate this next to where they are used instead of at the top of the file. They should have docs comments as well.